### PR TITLE
Only consider pseudo constants for detecting JOIN ON false

### LINKED ISSUE #4706: Local table with dist/ref with Gives Wrong Output on JOIN FALSE clauses
The output of the `SELECT` query changes after `create_reference_table`.
```
CREATE TABLE tbl1(a int);
CREATE TABLE tbl2(a int);

INSERT INTO tbl1 VALUES (1);
INSERT INTO tbl2 VALUES (1);

SELECT 1 AS res FROM tbl2 RIGHT JOIN (SELECT 1 FROM tbl1, tbl2) AS sub1 ON false;
┌─────┐
│ res │
├─────┤
│   1 │
└─────┘
(1 row)

SELECT create_reference_table('tbl2');
SELECT 1 AS res FROM tbl2 RIGHT JOIN (SELECT 1 FROM tbl1, tbl2) AS sub1 ON false;
┌─────┐ 
│ res │ 
├─────┤ 
└─────┘ 
(0 rows)
```

### PR DESCRIPTION
**Pull request by SaitTalhaNisanci**
It seems that we need to consider only pseudo constants while doing some
shortcuts in planning. For example there could be a false clause but it
can contribute to the result in which case it will not be a pseudo
constant.

Fixes #4706.

### CODE REVIEW

**Comment by SaitTalhaNisanci** [comment]
+DEBUG:  successfully loaded JIT provider in current session
+DEBUG:  time to inline: 0.000s, opt: 0.000s, emit: 0.003s

We might need to use notice log message. This is the first time I see such debug messages, we can also normalize them

**Review by onderkalaci — COMMENTED** [review]
I think in general we had a confusing between `where false` vs `join leads to one time filter False` in the code comments.

File: src/backend/distributed/planner/multi_router_planner.c [hunk 1]
```diff
@@ -2760,6 +2758,23 @@ TargetShardIntervalsForRestrictInfo(RelationRestrictionContext *restrictionConte
 }
 
 
+/*
+ * HasPseudoFalseClause returns true for queries that
```
[the remark this answers is not in the document]
Comment (SaitTalhaNisanci) [reply src/backend/distributed/planner/multi_router_planner.c:2762]: After renaming all this makes sense

File: src/backend/distributed/planner/multi_router_planner.c [hunk 2]
```diff
@@ -2705,16 +2705,15 @@ TargetShardIntervalsForRestrictInfo(RelationRestrictionContext *restrictionConte
...
-		List *pseudoRestrictionList = extract_actual_clauses(joinInfoList, true);
 
 		/*
 		 * Queries may have contradiction clauses like 'false', or '1=0' in
 		 * their filters. Such queries would have pseudo constant 'false'
 		 * inside relOptInfo->joininfo list. We treat such cases as if all
 		 * shards of the table are pruned out.
 		 */
-		bool whereFalseQuery = ContainsFalseClause(pseudoRestrictionList);
+		bool whereFalseQuery = JoinConditionIsOnFalse(
```
Comment (onderkalaci) [inline src/backend/distributed/planner/multi_router_planner.c:2715]: minor: `whereFalseQuery ` should be `joinFalseQuery`

**Review by onderkalaci — APPROVED** [review]

**Pull request merged**